src/main/java/com/artipie/http/PromuSlice.java

```diff
@@ -51,41 +53,51 @@ public PromuSlice(final Metrics metrics) {
     @Override
     public Response response(final String line, final Iterable<Map.Entry<String, String>> headers,
         final Publisher<ByteBuffer> body) {
-        final String ctype = new Accept(headers).values().get(0);
-        final StringWriter writer = new StringWriter();
-        final Set<String> filters = PromuSlice.params(line, "name");
-        final MetricsOutput output = new PrometheusOutput(writer, ctype, filters);
-        this.metrics.publish(output);
-        return new RsFull(
-            RsStatus.OK,
-            new Headers.From(new ContentType(ctype)),
-            new Content.From(
-                String.valueOf(writer.getBuffer()).getBytes(StandardCharsets.UTF_8)
-            )
-        );
+        final String ctype = new RqHeaders(headers, Accept.NAME).get(0);
+        try (StringWriter writer = new StringWriter()) {
+            final Set<String> filters = PromuSlice.params(line, "name");
+            final MetricsOutput output = new PrometheusOutput(writer, ctype, filters);
+            this.metrics.publish(output);
+            return new RsFull(
+                RsStatus.OK,
+                new Headers.From(new ContentType(ctype)),
+                new Content.From(
+                    String.valueOf(writer.getBuffer()).getBytes(StandardCharsets.UTF_8)
```

Comment:
You can just use `writer.toString().getBytes(...)` here, see https://docs.oracle.com/javase/7/docs/api/java/io/StringWriter.html#toString()

Reply:
Yeah, you are right.

Reply:
Yeah, you are right.